ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -484,6 +484,43 @@ interface {{ ethernet_interface.name }}
 {%         if ethernet_interface.pim.ipv4.dr_priority is arista.avd.defined %}
    pim ipv4 dr-priority {{ ethernet_interface.pim.ipv4.dr_priority }}
 {%         endif %}
+{%         if ethernet_interface.poe.priority is arista.avd.defined %}
+   poe priority {{ ethernet_interface.poe.priority }}
+{%         endif %}
+{%         if ethernet_interface.poe.reboot.action is arista.avd.defined %}
+   poe reboot action {{ ethernet_interface.poe.reboot.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.link_down.action is arista.avd.defined %}
+{%             set poe_link_down_action_cli = 'poe link down action ' ~ ethernet_interface.poe.link_down.action %}
+{%             if ethernet_interface.poe.link_down.power_off_delay is arista.avd.defined %}
+{%                 set poe_link_down_action_cli = poe_link_down_action_cli ~ ' ' ~ ethernet_interface.poe.link_down.power_off_delay %}
+{%             endif %}
+   {{ poe_link_down_action_cli }}
+{%         endif %}
+{%         if ethernet_interface.poe.shutdown.action is arista.avd.defined %}
+   poe shutdown action {{ ethernet_interface.poe.shutdown.action }}
+{%         endif %}
+{%         if ethernet_interface.poe.disabled is arista.avd.defined(true) %}
+   poe disabled
+{%         endif %}
+{%         if ethernet_interface.poe.limit.class is arista.avd.defined %}
```

Comment:
Insert an outer if to check for `ethernet_interface.poe.limit is arista.avd.defined`. Then you know that `poe_limit_cli` will be set when you print out or add `fixed`, so you can skip those ifs. It will also ensure that all the code will be skipped if not set.